### PR TITLE
add database infra class

### PR DESCRIPTION
**Pull request by evgenyud-ML**

### CODE REVIEW

File: data_preparation/database.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            else:
+                self.cursor.execute(query, params)
+
+            data = self.cursor.fetchall()
+            columns = [desc[0] for desc in self.cursor.description]
+            data_as_df = pd.DataFrame(data, columns=columns)
+
+        except Exception as e:
+            # todo change to log instead of print
+            print('select_query failed:\n Query: {}\n with Params: {}. Exception details: {}'.format(query, params, e))
```
Comment (helico) [inline data_preparation/database.py]: would be replaced by logger in the future

File: common/config.py [hunk 2]
```diff
@@ -13,7 +13,8 @@ def __init__(self, file_name='default_config', ignore_not_existed_files=True):
             dirname = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
             file_name = os.path.join(dirname, 'research_config.yml')
 
-        self._config = metayaml.read(file_name, ignore_not_existed_files=ignore_not_existed_files)
+        self._config = metayaml.read(file_name, defaults = {'env': os.environ},
```
Comment (helico) [inline common/config.py]: Didn't find the usage of this parameter

File: common/tests/test_database.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import unittest
+from data_preparation.database import ML_Database
+
+
+class TestDataBase(unittest.TestCase):
```
Comment (helico) [inline common/tests/test_database.py]: please separate 2 selection by 2 functions

Comment (helico) [reply common/tests/test_database.py]: Please generate test for internal exception case

File: data_preparation/database.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        """
+        connect to database and open cursor
+        if a connect cannot be made an exception will be raised here
+        """
+        self.ps_connection = psycopg2.connect(user=self.db_params['user'],
+                                              password=self.db_params['password'],
+                                              host=self.db_params['db_host'],
+                                              port=self.db_params['db_port'],
+                                              database=self.db_params['database'])
+        self.cursor = self.ps_connection.cursor()
```
[the remark this answers is not in the document]
Comment (evgenyud-ML) [reply data_preparation/database.py]: why ? 
its a private function, maybe rename it to start with 'underscore'. 
We will open connection while someone want to execute a query and close the connection immediately.

File: data_preparation/database.py (hunk 1 above)
Comment (evgenyud-ML) [reply data_preparation/database.py]: great!

File: common/config.py (hunk 2 above)
Comment (evgenyud-ML) [reply common/config.py]: which one ?

File: common/tests/test_database.py (hunk 3 above)
Comment (evgenyud-ML) [reply common/tests/test_database.py]: 1 done, 2 i will add more tests

**Pull request closed without merging**